src/interchange/avro.rs

```diff
@@ -481,29 +633,22 @@ impl Decoder {
         let writer_schemas =
             schema_registry.map(|sr| SchemaCache::new(sr, reader_schema.fingerprint::<Sha256>()));
 
-        let fast_row_schema = match reader_schema.top_node().inner {
-            // If the first two fields in the record are `before` and `after`,
-            // we don't need to decode the whole record. This can yield a
-            // substantial performance win when there is additional heavyweight
-            // metadata at the end of each record which would be immediately
-            // discarded.
-            SchemaPiece::Record { fields, .. }
-                if fields[0].name == "before" && fields[1].name == "after" =>
-            {
-                let node = SchemaNodeOrNamed {
-                    root: &reader_schema,
-                    inner: fields[0].schema.as_ref(),
-                };
-                Some(node.to_schema())
-            }
-            _ => None,
+        let debezium = if envelope == EnvelopeType::Debezium {
+            DebeziumDecodeState::new_from_schema(&reader_schema)
+        } else {
+            None
         };
-
+        // TODO - This should really cause `new` to fail, as the decoder is basically useless in this state.
+        // But before we can make `Decoder::new` return a `Result`, we first need to pipe the runtime errors
+        // stuff through and make avro decoding fallible.
+        if envelope == EnvelopeType::Debezium && debezium.is_none() {
+            error!("Couldn't extract Debezium schema information! No records will be decoded with this schema!");
+        }
```

Comment:
Feels like it would be cleaner to inline this in the above `if`? I spent a minute puzzling over this since on a quick skim it looks like the only way `debezium` can be `None` is if `envelope != Envelopetype::Debezium`.

```rust
let debezium = if envelope == EnvelopeType::Debezium {
    let ds = DebeziumDecodeState::new_from_schema(&reader_schema);
    if ds.is_none() { error!(...) }
    ds
} else {
    None
}
```

On second thought: can't we just perform this validation in `purify_statement`? And then just panic here if we don't have the Debezium fields? No RTEs necessary.